src/HDF5.jl

```diff
@@ -130,6 +130,7 @@ const H5I_DATASPACE    = 4
 const H5I_DATASET      = 5
 const H5I_ATTR         = 6
 const H5I_REFERENCE    = 7
+const H5I_VFL          = 8
```

Comment:
Should we just round out this enum and add

```
H5I_GENPROP_CLS = 8 
H5I_GENPROP_LST = 9  
H5I_ERROR_CLASS = 10 
H5I_ERROR_MSG = 11 
H5I_ERROR_STACK = 12 
H5I_NTYPES = 13 
``` 
?

Reply:
Sure, might as well.

Reply:
It looks like the enum changed in HDF5 v1.12, and my computer's header doesn't give the same numbering, and I assume you've got an off-by-one in the list you added, with the values supposed to be starting at 9? That would seem to match up with [this handy diff](https://fossies.org/diffs/hdf5/1.10.6_vs_1.12.0/src/H5Ipublic.h-diff.html) I found online

Reply:
They changed the enum in v1.12, well, let's not change this then until we sync up the BinaryBuilder hdf5 library.

Reply:
Sorry, I'm not quite sure what you're stuggesting — are you saying not to add just the extra ones you suggested, or you'd like to not add the `H5I_VFL` constant as well? I added just the one value because I was trying to test the `fapl[:driver]` property, and the actual return value is an even-harder-to-access constant than just verifying that it has the right type. (Though I suppose any non-erroring call is valid, so the test could just be `@test fapl[:driver] isa hid_t` and it'd probably be fine.)

Reply:
>  are you saying not to add just the extra ones you suggested,

Yes to this. Ignore my suggestion to add the extra ones.